### PR TITLE
[bgp]: Update FRR to version 7.4

### PR DESCRIPTION
**Pull request by pavel-shirshov**
**- Why I did it**
I updated frr version to latest stable. The latest stable version has more features we need (EVPN Type 5), also as will be supported longer.

**- How I did it**
First I updated sonic-frr repo and then updated our frr patches, and build system to use version 7.4.

**- How to verify it**
Build an image and run it on your DUT

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [x] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by pavel-shirshov** [comment]
retest vs please

**Comment by pavel-shirshov** [comment]
retest mellanox please

**Comment by pavel-shirshov** [comment]
retest baseimage please

File: dockers/docker-fpm-frr/frr/bgpd/bgpd.main.conf.j2 [hunk 1]
```diff
@@ -41,6 +41,7 @@ router bgp {{ DEVICE_METADATA['localhost']['bgp_asn'] }}
 {% block bgp_init %}
   bgp log-neighbor-changes
   no bgp default ipv4-unicast
+  no bgp ebgp-requires-policy
```
Comment (lguohan) [inline dockers/docker-fpm-frr/frr/bgpd/bgpd.main.conf.j2:44]: what is this policy?

Comment (pavel-shirshov) [reply dockers/docker-fpm-frr/frr/bgpd/bgpd.main.conf.j2:44]: http://docs.frrouting.org/en/latest/bgp.html?highlight=ebgp-policy#require-policy-on-ebgp
incoming and outcoming filters

**Comment by pavel-shirshov** [comment]
vsimage tests returns:
```
[2020-10-01T15:25:49.816Z] bgp/test_bgp_gr_helper.py::test_bgp_gr_helper_routes_perserved 
[2020-10-01T15:25:49.816Z] -------------------------------- live log call ---------------------------------
[2020-10-01T15:25:49.816Z] 15:25:41 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:25:49.816Z] 15:25:47 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:25:54.019Z] 15:25:53 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:25:59.323Z] 15:25:59 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:26:05.564Z] 15:26:05 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:26:13.528Z] 15:26:11 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:26:17.800Z] 15:26:17 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:26:24.372Z] 15:26:23 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:26:31.529Z] 15:26:29 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:26:35.731Z] 15:26:35 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
[2020-10-01T15:26:43.949Z] 15:26:41 ERROR utilities.py:wait_until:39: Exception caught while checking check_bgp_session_nsf: KeyError('bgpStateIs',)
```

**Comment by lguohan** [comment]
can you have more description of the pr?

**Comment by lguohan** [comment]
not sure about the vsimage tests returns, can you explain?

**Comment by pavel-shirshov** [comment]
vsimage tests failed before in this test [test_bgp_gr_helper.py](https://github.com/Azure/sonic-mgmt/blob/master/tests/bgp/test_bgp_gr_helper.py#L79 ) with the message: `Failed: neighbor fc00::7e does not enter NSF state`
I checked the code of the test and found this will show if [check_bgp_session_nsf()](https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices.py#L840) return `False`
I investigate the output of the test and found that `bgpStateIs` is not in the output.
I checked frr code and found that it is possible to skip it [bg[_vty.c](https://github.com/FRRouting/frr/blob/stable/7.4/bgpd/bgp_vty.c#L10852)
But it also possible that `bgpState` is not in `Active` state too. So I think it is worth to restart vsimage test again

**Comment by pavel-shirshov** [comment]
retest vsimage

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
I made the test manually and I don't understand why vsimage is failing. I checked the test on both ipv4 and ipv6 and both of them show passiveNSF
```
admin@str-s6100-acs-1:~$ vtysh -c 'show bgp ipv6 neighbors fc00::1a json'
{
  "fc00::1a":{
    "remoteAs":64802,
    "localAs":64601,
    "nbrExternalLink":true,
    "nbrDesc":"ARISTA04T1",
    "peerGroup":"PEER_V6",
    "bgpVersion":4,
    "remoteRouterId":"0.0.0.0",
    "localRouterId":"10.1.0.32",
    "bgpState":"Active",
    "bgpStateIs":"passiveNSF",
    "bgpTimerLastRead":29000,
    "bgpTimerLastWrite":29000,
    "bgpInUpdateElapsedTimeMsecs":145000,
    "bgpTimerHoldTimeMsecs":10000,
    "bgpTimerKeepAliveIntervalMsecs":3000,
    "bgpTimerConfiguredHoldTimeMsecs":10000,
    "bgpTimerConfiguredKeepAliveIntervalMsecs":3000,
    "gracefulRestartInfo":{
      "endOfRibSend":{
      },
      "endOfRibRecv":{
      },
      "gracefulRestartTimerMsecs":273000,
      "gracefulStalepathTimerMsecs":333000,
      "localGrMode":"Restart*",
      "remoteGrMode":"NotApplicable",
      "rBit":false,
      "timers":{
        "configuredRestartTimer":240,
        "receivedRestartTimer":300,
        "restartTimerRemaining":273
      }
    },
    "messageStats":{
      "depthInq":0,
      "depthOutq":0,
      "opensSent":2,
      "opensRecv":1,
      "notificationsSent":2,
      "notificationsRecv":0,
      "updatesSent":6,
      "updatesRecv":3,
      "keepalivesSent":11,
      "keepalivesRecv":12,
      "routeRefreshSent":0,
      "routeRefreshRecv":0,
      "capabilitySent":0,
      "capabilityRecv":0,
      "totalSent":21,
      "totalRecv":16
    },
    "minBtwnAdvertisementRunsTimerMsecs":0,
    "addressFamilyInfo":{
      "ipv6Unicast":{
        "peerGroupMember":"PEER_V6",
        "inboundSoftConfigPermit":true,
        "commAttriSentToNbr":"extendedAndStandard",
        "inboundPathPolicyConfig":true,
        "outboundPathPolicyConfig":true,
        "routeMapForIncomingAdvertisements":"FROM_BGP_PEER_V6",
        "routeMapForOutgoingAdvertisements":"TO_BGP_PEER_V6",
        "acceptedPrefixCounter":1
      }
    },
    "connectionsEstablished":1,
    "connectionsDropped":1,
    "lastResetTimerMsecs":26000,
    "lastResetDueTo":"NSF peer closed the session",
    "lastResetCode":22,
    "connectRetryTimer":120,
    "readThread":"off",
    "writeThread":"off"
  }
}
```

**Review by hasan-brcm — APPROVED** [review]

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by pavel-shirshov** [comment]
retest vsimage please

**Comment by wendani** [comment]
retest vsimage please

**Comment by lguohan** [comment]
retest vsimage please

**Pull request closed without merging**